packages/ember-metal/lib/expand_properties.js

```diff
@@ -8,6 +8,27 @@ import { assert } from './debug';
 const SPLIT_REGEX = /\{|\}/;
 const END_WITH_EACH_REGEX = /\.@each$/;
 
+// Checks if the braces inside a properties path are not nested and balanced
+function checkPropertiesBalance(str) {
```

Comment:
Can you move this down to the assertion so that it will also be stripped from production builds?

```js
assert(
  'Brace expanded properties have to be balanced and cannot be nested, e.g. "user.{name.{first,last},age}" and "user.{firstName},lastName}" are not allowed',
  ((str) => {
    let inBrace = 0;
    let char;
    for (let i = 0; i < str.length; i++) {
      char = str.charAt(i);

      if (char === '{') {
        inBrace++;
      } else if (char === '}') {
        inBrace--;
      }

      if (inBrace > 1 || inBrace < 0) {
        return false;
      }
    }

    return true;
  })(pattern)
);
```